stdlib/public/core/Character.swift

```diff
@@ -203,7 +202,8 @@ public struct Character :
   @_versioned
   internal init(_largeRepresentationString s: String) {
     if let native = s._core.nativeBuffer,
-       native.start == s._core._baseAddress! {
+      native.start == s._core._baseAddress!,
+      native.usedCount == s._core.count {
       _representation = .large(native._storage)
```

Comment:
So in theory, until we reform Character's small representation to be inline UTF16, will this pessimize code that iteratively peeks at the first character of a String/Substring? That is, are we speeding up String construction from a single Character at the cost of non-ASCII consumer loops?